randovania/interface_common/world_database.py

```diff
@@ -85,36 +87,42 @@ def __init__(self, persist_path: Path):
         self._all_data = {}
         self._lock = asyncio.Lock()
 
-    async def _read_data(self, uid: uuid.UUID) -> WorldData | None:
+    async def _read_data(self, path: Path) -> WorldData | None:
         try:
             raw_data = typing.cast(
                 "dict",
-                await json_lib.read_path_async(self._persist_path.joinpath(f"{uid}.json")),
+                await json_lib.read_path_async(path),
             )
             return WorldData.from_json(migrate_to_current(raw_data)["data"])
+        except UnsupportedVersion:
+            return None
         except json.decoder.JSONDecodeError:
-            self._persist_path.joinpath(f"{uid}.json").unlink()
+            path.unlink()
             return None
 
     async def _write_data(self, uid: uuid.UUID, data: WorldData):
         json_lib.write_path(
-            self._persist_path.joinpath(f"{uid}.json"),
+            self._persist_path.joinpath(str(CURRENT_VERSION), f"{uid}.json"),
             {
                 "schema_version": CURRENT_VERSION,
                 "data": data.as_json,
             },
         )
 
-    async def load_existing_data(self):
-        for f in self._persist_path.glob("*.json"):
+    async def load_existing_data(self) -> None:
+        paths = [self._persist_path]
+        for version in range(2, CURRENT_VERSION + 1):  # versioned folder started at 2
+            paths.append(self._persist_path.joinpath(str(version)))
+
+        for f in itertools.chain.from_iterable([path.glob("*.json") for path in reversed(paths)]):
```

Comment:
since its iterating through all data in reverse, does this not cause a scenario where e.g. different versions have the same world uuids, causing rdv to only read the earlier one?